services/galley/test/integration/API/Util.hs

```diff
@@ -1671,19 +1671,19 @@ withTempMockFederator ::
   (MonadIO m, ToJSON a) =>
   Opts.Opts ->
   Domain ->
-  a ->
+  (FederatedRequest -> a) ->
```

Comment:
I think I don't quite understand what this does and how the `resp / Request -> a` argument should be used. Could you add some inline comments here to these arguments?